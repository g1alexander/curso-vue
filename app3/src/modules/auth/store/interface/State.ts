export interface AuthState {
  status: string;
  user: null | User;
  idtoken: null | string;
  refreshtoken: null | string;
}

interface User {
  email: string;
  name: string;
}
