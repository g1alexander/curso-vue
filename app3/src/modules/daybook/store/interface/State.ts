export interface JournalState {
  isLoading: boolean;
  entries: Entry[];
}

interface Entry {
  id: number;
  date: string;
  text: string;
  picture: string | null;
}
