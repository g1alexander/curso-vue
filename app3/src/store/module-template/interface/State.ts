// export interface DayBookState {
//   count: number;
// }
