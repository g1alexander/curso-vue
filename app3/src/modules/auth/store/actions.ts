import { State } from "@/store";
import { ActionContext } from "vuex";
import { AuthState } from "./interface/State";
import { authApi } from "../../../api/authApi";
import { AxiosError } from "axios";

type Context = ActionContext<AuthState, State>;

// export const myAction = async (store: Context): Promise<void> => {
//   console.log(store);
// };

interface User {
  name: string;
  email: string;
  password?: string;
}

export const createUser = async (
  store: Context,
  user: User
): Promise<{ ok: boolean; message: string }> => {
  const { email, name, password } = user;

  try {
    const { data } = await authApi.post(":signUp", {
      email,
      password,
      returnSecureToken: true,
    });

    const { idToken, refreshToken } = data;

    await authApi.post(":update", {
      displayName: name,
      idToken,
    });

    delete user.password;
    store.commit("loginUser", { user, idToken, refreshToken });

    return { ok: true, message: "ok" };
  } catch (error: unknown) {
    const err = error as AxiosError;

    return { ok: false, message: err.response?.data.error.message };
  }
};

export const signInUser = async (
  store: Context,
  user: User
): Promise<{ ok: boolean; message: string }> => {
  const { email, password } = user;

  try {
    const { data } = await authApi.post(":signInWithPassword", {
      email,
      password,
      returnSecureToken: true,
    });

    const { idToken, refreshToken, displayName } = data;

    user.name = displayName;

    delete user.password;

    store.commit("loginUser", { user, idToken, refreshToken });

    return { ok: true, message: "ok" };
  } catch (error: unknown) {
    const err = error as AxiosError;

    return { ok: false, message: err.response?.data.error.message };
  }
};
