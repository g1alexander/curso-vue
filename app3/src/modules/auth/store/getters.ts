import { AuthState } from "./interface/State";

// export const myGetters = (state: AuthState) => {
//   return;
// };
export const currentState = (state: AuthState) => {
  return state.status;
};

export const username = (state: AuthState) => {
  return state.user?.name || "";
};
