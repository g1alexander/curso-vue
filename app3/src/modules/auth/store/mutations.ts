import { AuthState } from "./interface/State";

// export const myMutations = (state: AuthState) => {
//   console.log(state);
// };

interface Data {
  user: User;
  idToken: string;
  refreshToken: string;
}

interface User {
  email: string;
  name: string;
}

export const loginUser = (state: AuthState, data: Data) => {
  if (data.idToken) {
    localStorage.setItem("idToken", data.idToken);
    state.idtoken = data.idToken;
  }

  if (data.refreshToken) {
    localStorage.setItem("refreshToken", data.refreshToken);
    state.refreshtoken = data.refreshToken;
  }

  state.user = data.user;
  state.status = "authenticated";
};

export const logout = (state: AuthState) => {
  state.user = null;
  state.status = "not-authenticated";
  state.idtoken = null;
  state.refreshtoken = null;

  localStorage.removeItem("idToken");
  localStorage.removeItem("refreshToken");
};
