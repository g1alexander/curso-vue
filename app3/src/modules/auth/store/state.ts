import { AuthState } from "./interface/State";

const state: AuthState = {
  status: "authenticating",
  user: null,
  idtoken: null,
  refreshtoken: null,
};

export default (): AuthState => ({ ...state });
