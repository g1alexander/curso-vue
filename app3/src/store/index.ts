// store.ts
import { JournalState } from "@/modules/daybook/store/interface/State";
import { InjectionKey } from "vue";
import { createStore, useStore as baseUseStore, Store } from "vuex";
import { journalModule } from "@/modules/daybook/store/";

export interface State {
  journalModule: JournalState;
}

export const key: InjectionKey<Store<State>> = Symbol();

export const store = createStore<State>({
  modules: {
    journalModule,
  },
});

// define your own `useStore` composition function
export function useStore(): Store<State> {
  return baseUseStore(key);
}
