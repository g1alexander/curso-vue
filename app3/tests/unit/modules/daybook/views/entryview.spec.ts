import { shallowMount, VueWrapper } from "@vue/test-utils";
import EntryViewVue from "@/modules/daybook/views/EntryView.vue";

import { createStore } from "vuex";
import { journalModule } from "@/modules/daybook/store";
import { JournalState } from "@/modules/daybook/store/interface/State";
import { journalState } from "../../../mocks/journal-state";

const createVuexStore = (initialState: JournalState) =>
  createStore({
    modules: {
      journalModule: {
        ...journalModule,
        state: { ...initialState },
      },
    },
  });

describe("tests componet EntryView", () => {
  const $router = {
    push: jest.fn(),
  };

  const store = createVuexStore(journalState);

  let wrapper = shallowMount(EntryViewVue, {
    props: { id: "-MwZ3AI4JD9U9p5PJ3ay" },
    global: { plugins: [store], mocks: { $router } },
  });

  beforeEach(() => {
    jest.clearAllMocks();

    wrapper = shallowMount(EntryViewVue, {
      props: { id: "-MwZ3AI4JD9U9p5PJ3ay" },
      global: { plugins: [store], mocks: { $router } },
    });
  });

  test("debe de sacar al usuario si no esta el id", () => {
    const wrapper = shallowMount(EntryViewVue, {
      props: { id: "NO EXISTE" },
      global: { plugins: [store], mocks: { $router } },
    });

    expect(wrapper.vm.$router.push).toHaveBeenCalledWith({ name: "no-entry" });
  });

  test("debe de mostrar la entrada correctamente", () => {
    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.vm.$router.push).not.toHaveBeenCalled();
  });
});
