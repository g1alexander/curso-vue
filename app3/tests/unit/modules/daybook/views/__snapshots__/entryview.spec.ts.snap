// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests componet EntryView debe de mostrar la entrada correctamente 1`] = `
<div class="entry-title d-flex justify-content-between p-2">
  <div><span class="text-success fd-3 fw-bold">22</span><span class="mx-1 fs-3">Marzo</span><span class="mx-2 fs-4 fw-light">2022, Martes</span></div>
  <div><button class="btn btn-danger mx-2"> Borrar <i class="fa fa-trash-alt"></i></button><input type="file" style="display: none;"><button class="btn btn-primary mx-2"> Subir <i class="fa fa-upload"></i></button></div>
</div>
<hr>
<div class="d-flex flex-column w-100 px-3 h-75"><textarea placeholder="¿Que sucedió hoy?"></textarea></div><img src="https://res.cloudinary.com/dlgvxohur/image/upload/v1645577816/journal/mrlu5sfmsyrkcozpyciy.jpg" class="img-thumbanail">
<!--v-if-->
<fab-stub icon="fa-save"></fab-stub>
`;
